File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int s=0, e=matrix[0].length-1;
        while(s<matrix.length && e>=0){
            if(target==matrix[s][e])return true;
            else if(target>matrix[s][e])s++;
            else e--;
        }
        return false;
    }
}

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        s=0
        e=len(matrix[0])-1
        while(s<len(matrix) and e>=0):
            if target==matrix[s][e]:
                return True
            elif target>matrix[s][e]:
                s+=1
            else:
                e-=1
        return False